### PR TITLE
Fix ENS decoder initialization and avatar queries

### PR DESCRIPTION
**Pull request by LefterisJP**

### CODE REVIEW

**Comment by LefterisJP** [comment]
The globalDB has the following additions:

```sql
sqlite> INSERT INTO contract_abi(value, name) VALUES('[{"inputs":[{"internalType":"contract ENS","name":"_old","type":"address"}],"payable":false,"stateMutability":"nonpayable","type":"constructor"},{"anonymous":false,"inputs":[{"indexed":true,"internalType":"address","name":"owner","type":"address"},{"indexed":true,"internalType":"address","name":"operator","type":"address"},{"indexed":false,"internalType":"bool","name":"approved","type":"bool"}],"name":"ApprovalForAll","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"internalType":"bytes32","name":"node","type":"bytes32"},{"indexed":true,"internalType":"bytes32","name":"label","type":"bytes32"},{"indexed":false,"internalType":"address","name":"owner","type":"address"}],"name":"NewOwner","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"internalType":"bytes32","name":"node","type":"bytes32"},{"indexed":false,"internalType":"address","name":"resolver","type":"address"}],"name":"NewResolver","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"internalType":"bytes32","name":"node","type":"bytes32"},{"indexed":false,"internalType":"uint64","name":"ttl","type":"uint64"}],"name":"NewTTL","type":"event"},{"anonymous":false,"inputs":[{"indexed":true,"internalType":"bytes32","name":"node","type":"bytes32"},{"indexed":false,"internalType":"address","name":"owner","type":"address"}],"name":"Transfer","type":"event"},{"constant":true,"inputs":[{"internalType":"address","name":"owner","type":"address"},{"internalType":"address","name":"operator","type":"address"}],"name":"isApprovedForAll","outputs":[{"internalType":"bool","name":"","type":"bool"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[],"name":"old","outputs":[{"internalType":"contract ENS","name":"","type":"address"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"internalType":"bytes32","name":"node","type":"bytes32"}],"name":"owner","outputs":[{"internalType":"address","name":"","type":"address"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"internalType":"bytes32","name":"node","type":"bytes32"}],"name":"recordExists","outputs":[{"internalType":"bool","name":"","type":"bool"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":true,"inputs":[{"internalType":"bytes32","name":"node","type":"bytes32"}],"name":"resolver","outputs":[{"internalType":"address","name":"","type":"address"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":false,"inputs":[{"internalType":"address","name":"operator","type":"address"},{"internalType":"bool","name":"approved","type":"bool"}],"name":"setApprovalForAll","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"},{"constant":false,"inputs":[{"internalType":"bytes32","name":"node","type":"bytes32"},{"internalType":"address","name":"owner","type":"address"}],"name":"setOwner","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"},{"constant":false,"inputs":[{"internalType":"bytes32","name":"node","type":"bytes32"},{"internalType":"address","name":"owner","type":"address"},{"internalType":"address","name":"resolver","type":"address"},{"internalType":"uint64","name":"ttl","type":"uint64"}],"name":"setRecord","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"},{"constant":false,"inputs":[{"internalType":"bytes32","name":"node","type":"bytes32"},{"internalType":"address","name":"resolver","type":"address"}],"name":"setResolver","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"},{"constant":false,"inputs":[{"internalType":"bytes32","name":"node","type":"bytes32"},{"internalType":"bytes32","name":"label","type":"bytes32"},{"internalType":"address","name":"owner","type":"address"}],"name":"setSubnodeOwner","outputs":[{"internalType":"bytes32","name":"","type":"bytes32"}],"payable":false,"stateMutability":"nonpayable","type":"function"},{"constant":false,"inputs":[{"internalType":"bytes32","name":"node","type":"bytes32"},{"internalType":"bytes32","name":"label","type":"bytes32"},{"internalType":"address","name":"owner","type":"address"},{"internalType":"address","name":"resolver","type":"address"},{"internalType":"uint64","name":"ttl","type":"uint64"}],"name":"setSubnodeRecord","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"},{"constant":false,"inputs":[{"internalType":"bytes32","name":"node","type":"bytes32"},{"internalType":"uint64","name":"ttl","type":"uint64"}],"name":"setTTL","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"},{"constant":true,"inputs":[{"internalType":"bytes32","name":"node","type":"bytes32"}],"name":"ttl","outputs":[{"internalType":"uint64","name":"","type":"uint64"}],"payable":false,"stateMutability":"view","type":"function"}]', 'ENS_REGISTRY');
```
This has id 78. And then added the actual contract too.

```sql
INSERT INTO contract_data(address,chain_id,abi,deployed_block) VALUES('0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e',1,78,9380380);
```

File: rotkehlchen/chain/ethereum/node_inquirer.py [hunk 1]
```diff
@@ -251,6 +232,41 @@ def _ens_lookup(
...
+            arguments=[normal_name_to_hash(normal_name)],
+        )
+        if is_none_or_zero_address(resolver_addr):
+            return None, None
+
+        try:
+            deserialized_resolver_addr = deserialize_evm_address(resolver_addr)
+        except DeserializationError:
+            log.error(
+                f'Error deserializing address {resolver_addr} while doing'
```
Comment (yabirgb) [inline rotkehlchen/chain/ethereum/node_inquirer.py:263]:
```suggestion
                f'Error deserializing address {resolver_addr} while doing '
```

File: rotkehlchen/tests/unit/test_evm_names.py [hunk 2]
```diff
@@ -132,22 +134,34 @@ def test_download_ens_avatar(ethereum_inquirer):
...
+        assert dbens.get_last_avatar_update('nebolax.eth') <= ts_now(), 'Last update timestamp should have been set'  # noqa: E501
+        try_download_ens_avatar(
+            eth_inquirer=ethereum_inquirer,
+            avatars_dir=tempdir,
+            ens_name='tewshi.eth',  # an avatar should be downloaded. Resolver v3
         )
-        assert dbens.get_last_avatar_update('nebolax.eth') == ts_now(), 'Last update timestamp should have been set'  # noqa: E501
-        assert list(tempdir.iterdir()) == [tempdir / 'nebolax.eth.png']
+        assert dbens.get_last_avatar_update('tewshi.eth') <= ts_now(), 'Last update timestamp should have been set'  # noqa: E501
+        assert list(tempdir.iterdir()) == [tempdir / 'tewshi.eth.png', tempdir / 'nebolax.eth.png']
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/test_evm_names.py:164]: Did you run it locally? Wonder why the order is different if so

File: rotkehlchen/chain/ethereum/node_inquirer.py (hunk 1 above)
Comment (LefterisJP) [reply rotkehlchen/chain/ethereum/node_inquirer.py:263]: Dude why did you not press review 10 secs later? I just pushed.

File: rotkehlchen/tests/unit/test_evm_names.py (hunk 2 above)
Comment (LefterisJP) [reply rotkehlchen/tests/unit/test_evm_names.py:164]: Becuse it's a different system and order of files is not guaranteed by iterdir

File: rotkehlchen/chain/ethereum/node_inquirer.py (hunk 1 above)
Comment (LefterisJP) [reply rotkehlchen/chain/ethereum/node_inquirer.py:263]: ah wait no I did not push. AHAHAAAAHAHAAHAH

**Pull request merged**